homeassistant/components/smartthings/lock.py

```diff
@@ -0,0 +1,54 @@
+"""
+Support for locks through the SmartThings cloud API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/smartthings.lock/
+"""
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import STATE_LOCKED
+
+from . import SmartThingsEntity
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add locks for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsLock(device) for device in broker.devices.values()
+         if is_lock(device)])
+
+
+def is_lock(device):
+    """Determine if the device supports the lock capability."""
+    from pysmartthings import Capability
+    if Capability.lock in device.capabilities:
+        return True
+    return False
+
+
+class SmartThingsLock(SmartThingsEntity, LockDevice):
```

Comment:
I think we're missing the supported features property.

Reply:
Not sure if it is a requirement, unless we are wanting to support lock devices which also provide control over automated door mechanisms at this time (as that's the only extra HASS feature that can be returned by that function). In SmartThings it seems to be handled by the Door Control capability.

Reply:
You're right!  We'd want to represent `doorControl` and `windowShade` with the cover platform as they have open/close.